nrf_security/doc/backend_config.rst

```diff
@@ -1,73 +1,43 @@
-.. _nrf_security_backend_config:
+.. _nrf_security_legacy_config:
 
-Backend configurations and supported features
-#############################################
+Legacy configurations and supported features
+############################################
 
 .. contents::
    :local:
    :depth: 2
 
-This section covers the configurations available when either linking directly to a backend library or when using the Mbed TLS glue layer.
+This section covers the configurations available when using legacy crypto API support.
 
 .. _nrf_security_backend_config_multiple:
 
-Configuring multiple backends
-*****************************
-
-Different backends support different cryptographic algorithms.
-The algorithms listed in subsequent sections are implemented either as *Glue* or *Choice*.
+Configuring backends
+********************
 
-Glue
-   Algorithms that can be enabled in multiple backends simultaneously.
-Choice
-   Algorithms that can not be enabled in multiple backends simultaneously.
-   To enable the algorithm, a `base` configuration must first be enabled, and then the backend is chosen using a Kconfig option prefixed with ``CONFIG_CHOICE_``.
-   If no backend is explicitly chosen it will be selected automatically with :ref:`nrf_security_backends_cc3xx` having precedence over :ref:`nrf_security_backends_oberon` and :ref:`nrf_security_backends_oberon` having precedence over :ref:`nrf_security_backends_orig_mbedtls`.
+The legacy configuration does not allow multiple backends being enabled at the same time. 
 
-It will be noted in the below tables when only some backends support a feature.
+The choice of implementation is controlled by setting :kconfig:`CONFIG_CC3XX_BACKEND` for devices with the CryptoCell hardware peripheral, or :konfig:`CONFIG_OBERON_BACKEND`.
```

Comment:
In this file we are mentioning "Original mBed TLS".

I think we should use a name that better communicates that we are not using vanilla mbedtls (it has patches).

Also, we should define what it is. E.g. that it is a Nordic fork of Zephyr's mbedtls fork and it is supplied by west at the directory ncs/mbedtls.

Unless this is documented elsewhere.

We could also mention that it is used to complement missing functionality in oberon and cryptocell.

Reply:
The code is original for all legacy APIs. Our changes are only affecting the PSA driver scope

But we can add a statement on our fork!